### PR TITLE
Slashing protection

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

> Uncomment one line below and remove others.
>
> Bug fix
 Feature
> Documentation
> Other

**What does this PR do? Why is it needed?**

bring changes on how importing files works
introduces an import keystore manager for single vs multi password
includes slashing protection import on non hd wallet and keystore imports.

**Which issues(s) does this PR fix?**

**Other notes for review**

### CODE REVIEW

File: src/app/modules/core/components/loading-overlay/loading-overlay.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div *ngIf="uploading">
+    <div class="text-white text-xl mt-4">
+      Uploading files...
+    </div>
+    <div class="my-4 text-hint text-lg leading-relaxed">
+      Hang in there while we upload your keystore files...
```
Comment (rauljordan) [inline src/app/modules/core/components/loading-overlay/loading-overlay.component.html:6]:
```suggestion
      Hang in there while we upload your files...
```
We could use this as an overlay for all kinds of things, not just keystores in the future

File: src/app/modules/onboarding/pages/nonhd-wallet-wizard/nonhd-wallet-wizard.component.ts [hunk 2]
```diff
@@ -99,29 +102,67 @@ export class NonhdWalletWizardComponent implements OnInit, OnDestroy {
...
   }
 
+  private get importKeystores$(): Observable<any>{
+    const keystoresImported: string[] = [];
+    let keystorePasswords: string[] = [];
+    (this.keystoresFormGroup.controls['keystoresImported'] as FormArray).controls.forEach((keystore: AbstractControl) => {
+      keystoresImported.push(JSON.stringify(keystore.get('keystore')?.value));
+      keystorePasswords.push(keystore.get('keystorePassword')?.value);
+    });
+    if(this.importAccounts?.uniqueToggleFormControl.value){
```
Comment (rauljordan) [inline src/app/modules/onboarding/pages/nonhd-wallet-wizard/nonhd-wallet-wizard.component.ts:117]: Might be worthwhile adding a comment about the complex observable logic this function is doing. We can just explain why we're doing a list of requests

File: src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.html [hunk 3]
```diff
@@ -1,63 +1,94 @@
...
+        <div *ngIf="!editMode" class="flex-none">
+          <button mat-raised-button color="primary" (click)="enterEditMode()">Edit</button>
+        </div>
+
+        <div *ngIf="editMode" class="flex-none"> 
+          <button mat-raised-button color="primary" (click)="removeKeystores()">Remove</button>
+          <button mat-raised-button (click)="editMode = false">Cancel</button>
+        </div>
+      </div>
+      <div class="overflow-y-auto keystore-list-wrapper">
```
Comment (rauljordan) [inline src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.html:38]: This is really intense! Great work on getting all of this done

File: src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.ts [hunk 4]
```diff
@@ -1,24 +1,72 @@
...
+  }
 
-  invalidFiles: string[] = [];
+  ngOnInit(): void {
+    this.uniqueToggleFormControl.valueChanges.subscribe((value) => {
+      this.keystorePasswordDefaultFormGroup.reset(this.keystorePasswordDefaultFormGroupInit);
+      this.keystoresImported.controls.forEach(fg => {
+        fg.get('keystorePassword')?.markAsPristine();
+      });
+      // really shitty to need this ... but angular doesn't detect changes fast enough after so we need to check for changes again...
```
Comment (rauljordan) [inline src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.ts:46]:
```suggestion
      // Angular doesn't detect changes fast enough after so we need to check for changes again...
```

File: src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.ts [hunk 5]
```diff
@@ -27,10 +75,19 @@ export class ImportAccountsFormComponent {
...
-            if (res) {
-              this.updateImportedKeystores(item, JSON.parse(res));
+          blob.forEach((item) => {
+            const file = blob.file(item);
+            if(file){
+              file.async('string').then((txt) => {
+                try {
+                this.updateImportedKeystores(item, JSON.parse(txt));
+                } catch(err){
+                  // sometimes zip contains extra info that processes like a file
```
Comment (rauljordan) [inline src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.ts:85]: Any log at least? What if there's an actual error

Comment (james-prysm) [reply src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.ts:85]: user experience wise it should just look like it didn't receive the files and it won't show up. I was thinking of logging but there would be a clutter or false positives I think...

File: src/app/modules/onboarding/pages/nonhd-wallet-wizard/nonhd-wallet-wizard.component.ts (hunk 2 above)
Comment (james-prysm) [reply src/app/modules/onboarding/pages/nonhd-wallet-wizard/nonhd-wallet-wizard.component.ts:121]: added some comments

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Great job! Really looking forward to this in the next release

**Pull request merged**